tests/integration/angular-meteor-collection-spec.js

```diff
@@ -115,4 +115,78 @@ describe('$meteorCollection service', function() {
       expect(meteorArray).toEqualCollection(MyCollection);
     });
   });
+
+  describe('observe server changes', function() {
+
+    beforeEach(function() {
+      // Cleaning up after each test.
+      bigCollection.find({}).forEach(function(doc) {
+        bigCollection.remove(doc._id);
+      });
+
+      for (var i = 0; i < 100; i++) {
+        bigCollection.insert({count: i});
+      }
+    });
+
+    it('$apply executed twice', function() {
+
+      var $ngCol = $meteorCollection(bigCollection);
+
+      expect($rootScope.$apply).toHaveBeenCalled();
+
+      expect($ngCol).toEqualCollection(bigCollection);
+
+      // No matter how much elements MyCollection contains
+      // $rootScope.$apply should be called twice.
+      expect($rootScope.$apply.calls.count()).toEqual(2);
+
+    });
+
+    it('push updates from client handled correctly', function() {
+
+      $rootScope.limit = 10;
+      var $ngCol = $meteorCollection(function() {
+        return bigCollection.find({}, {
+          limit: $rootScope.getReactively('limit')
+        });
+      });
+      spyOn($ngCol, 'save');
+
+      // Adds docs on the client.
+      for (var i = 100; i < 106; i++) {
+        $ngCol.push({count: i});
+      }
+
+      $timeout.flush();
+
+      expect($ngCol.length).toEqual(10);
+      expect($ngCol.save).toHaveBeenCalled();
+      expect($ngCol.save.calls.count()).toEqual(6);
```

Comment:
@barbatus 
That's very cool, on top of that, if might be good to check that bigCollection in the DB is now equal to 106